ansible/roles/debops.grafana/defaults/main.yml

```diff
@@ -0,0 +1,193 @@
+# used as the e-mail sender domain name.
+grafana__fqdn: 'grafana.{{ grafana__domain }}'
+
+                                                                   # ]]]
+# .. envvar:: grafana__domain [[[
+#
+# The DNS domain used by other variables in the ``debops.grafana`` role.
+grafana__domain: '{{ ansible_local.core.domain
+                    if (ansible_local|d() and ansible_local.core|d() and
+                        ansible_local.core.domain|d())
+                    else ansible_domain }}'
+
+                                                                   # ]]]
+                                                                   # ]]]
+# Grafana configuration [[[
+# --------------------------------
+
+# .. envvar:: grafana__nginx_enabled: True [[[
+#
+# Enable or disable Nginx dependencies.
+grafana__nginx_enabled: True
+
+                                                                   # ]]]
+# .. envvar:: grafana__upstream_apt_id [[[
+#
+# The Grafana fingerprint.
+grafana__upstream_apt_id: '4E40 DDF6 D76E 284A 4A67  80E4 8C8C 34C5 2409 8CB6'
+
+                                                                   # ]]]
+# .. envvar:: grafana__upstream_apt_repository [[[
+#
+# The APT repository URI of the upstream Grafana repository.
+grafana__upstream_apt_repository: 'deb https://packages.grafana.com/oss/deb stable main'
+
+                                                                   # ]]]
+                                                                   # ]]]
+# Installation, APT packages [[[
+# ------------------------------
+
+# .. envvar:: grafana__base_packages [[[
+#
+# List of APT packages which are required by the Grafana role.
+grafana__base_packages: [ 'grafana' ]
+
+                                                                   # ]]]
+# .. envvar:: grafana__packages [[[
+#
+# List of APT packages to install on all hosts in Grafana inventory.
+grafana__packages: [ 'software-properties-common', 'apt-transport-https' ]
+
+                                                                   # ]]]
+# .. envvar:: grafana__deploy_type [[[
+#
+# Select how Grafana should be installed by the role:
+#
+# - ``system``: install Grafana using current OS packages, possibly from the
+#               backports repository.
+#
+# - ``upstream``: install Grafana from the upstream APT repository on
+#                 Launchpad (PPA, usable on Debian as well).
+#
+# - ``bootstrap``: install APT packages required for Grafana and build a local
+#                  ``.deb`` package using the upstream GitHub repository.
+#
+grafana__deploy_type: "upstream"
+
+                                                                   # ]]]
+# .. envvar:: grafana__allow [[[
+#
+# List of IP addresses or CIDR subnets which will be allowed to connect to the
+# Grafana in :command:`ip(6)tables` and TCP wrappers. If it's empty, remote
+# connections are not allowed.
+grafana__allow: [ '0.0.0.0/0' ]
```

Comment:
Currently the value here is contradictory - it will allow access over IPv4 to the service from anywhere, but not over IPv6. And if the user will want to open the access over IPv6, he/she will have to copy over the `0.0.0.0/0` value as well because if not, it will not be present in overriden variable in the inventory and that will flip the IPv4/IPv6 access.

You should decide if access to Grafana should be public or private by default. Since this is a web service with authentication, I think that having it accessible by default through the firewall is OK. In that case, I would make the `grafana__allow` list empty, and change the `accept_any` parameter in `debops.ferm` configuration to `True` - this will allow access from anywhere by default and if users want to limit access to their network, they still can.

Reply:
modified in both places